hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ResolvedBucket.java

```diff
@@ -108,4 +117,25 @@ public boolean isLink() {
     return auditMap;
   }
 
+  /**
+   * Adds audit information about the bucket and its bucket layout (and if it's
+   * a link, then the real bucket, too) to {@code auditMap}.
+   *
+   * @return the same map for convenience
+   */
+  public Map<String, String> auditWithBucketLayout(Map<String, String> auditMap,
+                                                   OzoneManager om) {
+    audit(auditMap);
+    try {
+      // Add BucketLayout to auditMap from the resolved Bucket.
+      auditMap.put(OzoneConsts.BUCKET_LAYOUT, String.valueOf(
+          om.getBucketInfo(realVolume(), realBucket()).getBucketLayout()
```

Comment:
I would definitely avoid passing around OM and having to look up bucket info.

Reply:
I was skeptical of this approach as well but wasn't sure how else we could fetch the bucket layout. Since you already mentioned in another comment that we don't require this information added to the audit logs in except in OmBucketInfo - I will remove these methods and their usages.